core/src/test/java/org/apache/calcite/test/SqlValidatorTest.java

```diff
@@ -12167,4 +12167,48 @@ private void checkCustomColumnResolving(String table) {
         .withOperatorTable(operatorTable)
         .type("RecordType(BIGINT NOT NULL A, BIGINT B) NOT NULL");
   }
+
+  /**
+   * Tests that a named param can be typed in a select.
+   */
+  @Test void testSelectNamedParam() {
+    final String sql = "select @a, @b from emp";
+    sql(sql).withNamedParamters().ok();
+  }
+
+
+  /**
+   * Tests that a named param can be properly typed.
+   */
+  @Test void testBasicNamedParam() {
```

Comment:
Might want to include a test that checks that a typing error is properly thrown in the case that. I think we test for this on the BodoSql Side, so you can omit it if you don't want to fight the testing suite anymore.

Reply:
I'll file a JIRA to add this. I'm not sure how to reliably test this without significantly restructuring the test suite.

Reply:
Let's file a JIRA. I don't feel like fighting this anymore.

Reply:
https://bodo.atlassian.net/browse/BS-535